.github/copilot-instructions.md

```diff
@@ -0,0 +1,326 @@
+# iTwin.js Core - AI Coding Agent Instructions
```

Comment:
thx for creating this, this will be super useful. 
should we break this out into a separate pr?
would be good for leads like @pmconne @grigasp @ColinKerr @wgoehrig @dassaf4 to review

Reply:
sure.